src/main/java/core/basesyntax/figures/Rectangle.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.figures;
+
+import java.util.Locale;
+
+public class Rectangle extends Figure {
+    private double topSide;
+    private double lateralSide;
+    private double area;
+
+    public Rectangle(String color, double topSide, double lateralSide) {
+        super(color);
+        this.topSide = topSide;
+        this.lateralSide = lateralSide;
+        this.area = calculateArea();
+    }
+
+    public double getTopSide() {
+        return topSide;
+    }
+
+    public void setTopSide(double topSide) {
+        this.topSide = topSide;
+        this.area = calculateArea();
+    }
+
+    public double getLateralSide() {
+        return lateralSide;
+    }
+
+    public void setLateralSide(double lateralSide) {
+        this.lateralSide = lateralSide;
+        this.area = calculateArea();
```

Comment:
Similar to the setTopSide method, avoid recalculating the area and storing it as a field every time you set a new value for lateralSide. Calculate the area on-demand in the getArea() method.